src/plugins/mf_priority.cpp

```diff
@@ -252,19 +267,57 @@ static int validate_cb (flux_plugin_t *p,
     if (fairshare == 0)
         return flux_jobtap_reject_job (p, args, "user fairshare value is 0");
 
-    // make sure user has not already hit their max active jobs count
-    if (max_jobs > 0 && current_jobs >= max_jobs)
-        return flux_jobtap_reject_job (p, args,
-                                       "user has max number of jobs submitted");
-
     if (flux_jobtap_job_aux_set (p,
                                  FLUX_JOBTAP_CURRENT_JOB,
                                  "mf_priority:bank_info",
                                  &bank_it->second,
                                  NULL) < 0)
         flux_log_error (h, "flux_jobtap_job_aux_set");
 
-    bank_it->second.current_jobs++;
+    return 0;
+}
+
+
+static int depend_cb (flux_plugin_t *p,
+                      const char *topic,
+                      flux_plugin_arg_t *args,
+                      void *data)
+{
+    int userid;
+    long int id;
+    struct bank_info *b;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s:I}",
+                                "userid", &userid, "id", &id) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    b = static_cast<bank_info *> (flux_jobtap_job_aux_get (
+                                                    p,
+                                                    FLUX_JOBTAP_CURRENT_JOB,
+                                                    "mf_priority:bank_info"));
+
+    if (b == NULL)
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB, "plugin",
+                                     3, "mf_priority: bank info is missing");
+
```

Comment:
`return -1` if `b == NULL` to avoid NULL deref below.

Reply:
Thanks! Yeah, I think `return -1` would appropriately belong in both `job.state.priority` and `job.state.inactive`, so I'll go ahead and add those checks and probably put them in a separate commit. 

> Then, you may want to see if you can add these corner cases to the testsuite somehow to ensure the broker doesn't crash if a job from an unknown bank or other corner case is submitted.

I know for cases where:

- the user submitting the job doesn't belong in the flux-accounting DB, or 
- the user submits a job against a bank they don't belong to

The job gets rejected in the `job.validate` state before the struct is set and subsequently opened in the other `job.state` callbacks. These tests are handled in `t1001-mf-priority-basic.t`. Sending empty/incorrectly formatted payloads to the plugin are also tested in `t1001-mf-priority-basic.t`. 

I'll keep thinking about it and see if there's a corner case or scenario I'm missing.

Reply:
what about a case where a job is submitted while the bank/user exists, but then before the job reaches one of the `job.state` callbacks the user/bank pair is removed by an update? (Is that even possible?)

Even if that is not possible, it would be good to enable that in testing to get better coverage and ensure the plugin is well behaved.